### PR TITLE
[CI] automate update spirv-llvm-tranlsator commit

### PR DESCRIPTION
**Pull request by AshburnLee**
The core Triton is a small number of people, and we receive many PRs (thank
you!).  To help us review your code more quickly, **if you are a new
contributor (less than 3 PRs merged) we ask that you complete the following
tasks and include the filled-out checklist in your PR description.**

Complete the following tasks before sending your PR, and replace `[ ]` with
`[x]` to indicate you have done them.

- [x] I am not making a trivial change, such as fixing a typo in a comment.

- [x] I have written a PR description following these
  [rules](https://cbea.ms/git-commit/#why-not-how).

- [x] I have run `pre-commit run --from-ref origin/main --to-ref HEAD`.

- Select one of the following.
  - [ ] I have added tests.
    - `/test` for `lit` tests
    - `/unittest` for C++ tests
    - `/python/test` for end-to-end tests
  - [x] This PR does not need a test because `workflow for CI`.

- Select one of the following.
  - [x] I have not added any `lit` tests.
  - [ ] The `lit` tests I have added follow these [best practices](https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices),
    including the "tests should be minimal" section. (Usually running Python code
    and using the instructions it generates is not minimal.)

## Functionality
Fix https://github.com/intel/intel-xpu-backend-for-triton/issues/1672

### CODE REVIEW

File: .github/workflows/update-spirv-llvm-translator.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+#   schedule:
+#     - cron: "0 0 * * *"
+  pull_request:
+    branches:
+      - lijunhui/translator_cid
+  push:
+    branches:
+      - lijunhui/translator_cid
+# env:
+#   TORCH_XPU_OPS_COMMIT: 39522db63ce045f52c9d61a286018c266cd00479
```
Comment (pbchekin) [inline .github/workflows/update-spirv-llvm-translator.yml:13]: The comment is not needed.

File: .github/workflows/update-spirv-llvm-translator.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+      - name: Install test dependencies
+        run: |
+          pip install pytest pytest-xdist pytest-rerunfailures pytest-select pytest-timeout expecttest
+          pip install git+https://github.com/kwasd/pytest-capturewarnings-ng.git@v1.2.0
+
+      # - name: Setup Triton
+      #   uses: ./.github/actions/setup-triton
+      #   with:
+      #     build_llvm: false
```
Comment (pbchekin) [inline .github/workflows/update-spirv-llvm-translator.yml:58]: The comment is not needed.

File: .github/workflows/update-spirv-llvm-translator.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        id: translator-latest-commit
+        run: |
+          echo "cid_latest=$(git -C external/SPIRV-LLVM-Translator rev-parse HEAD)" >> $GITHUB_ENV
+      # - name: Run core tests
+      #   run: |
+      #     source ./scripts/pytest-utils.sh
+      #     ensure_spirv_dis
+      #     cd python/test/unit
+      #     TRITON_TEST_SUITE=language \
+      #       pytest -vvv -n 8 --device xpu language/ --ignore=language/test_line_info.py --ignore=language/test_subprocess.py
```
Comment (pbchekin) [inline .github/workflows/update-spirv-llvm-translator.yml:82]: The comment is not needed.

File: scripts/check_update_tranlator_cid.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+# check every commit ids
+cd $TRITON_PROJ
+for cid in $COMMIT_IDS; do
+    echo "$cid" > ./lib/Target/SPIRV/spirv-llvm-translator.conf
+    ./scripts/compile-triton.sh --clean
+    COMPILE_STATUS=$?
+    if [ $COMPILE_STATUS -ne 0 ]; then
+        echo "Compile failed for translator commit $cid"
+        continue
+    fi
```
Comment (pbchekin) [inline scripts/check_update_tranlator_cid.sh:35]:
```suggestion
    if ! ./scripts/compile-triton.sh --clean; then
        echo "Compile failed for translator commit $cid"
        continue
    fi
```

File: scripts/check_update_tranlator_cid.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    # build pytorch and IPEX outside of 'test-triton.sh'
+    ./scripts/test-triton.sh --skip-deps
+    TEST_STATUS=$?
+    if [ $TEST_STATUS -eq 0 ]; then
+        echo "Tests passed for translator commit $cid"
+        echo "A newer commit found: $cid"
+        break
+    else
+        echo "Tests failed for translator commit $cid"
+    fi
```
Comment (pbchekin) [inline scripts/check_update_tranlator_cid.sh:46]:
```suggestion
    if ./scripts/test-triton.sh --skip-deps; then
        echo "Tests passed for translator commit $cid"
        echo "A newer commit found: $cid"
        break
    else
        echo "Tests failed for translator commit $cid"
    fi
```

**Comment by AshburnLee** [comment]
This PR outputs a message: A newer commit found: f0d7e476924b8f9dad0428682b5633e908a43dde
I think this can be the target of this PR, right?

BTW, There was a merge from the upstream, so now the translator cid is the newest.

File: .github/workflows/update-spirv-llvm-translator.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    branches:
+      - lijunhui/translator_cid
+  push:
+    branches:
+      - lijunhui/translator_cid
+
+jobs:
+  update-translator-commit:
+    runs-on:
+      - rolling
```
[the remark this answers is not in the document]
Comment (AshburnLee) [reply .github/workflows/update-spirv-llvm-translator.yml]: I narrow down runners by adding more labels like some other workflows do.
Thanks.

**Review by pbchekin — APPROVED** [review]

**Pull request merged**